impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -296,20 +298,23 @@ public Map<String, Object> getConnectConfigForCluster(String bootstrapServers, S
             kafkaConfig.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, securityProtocol);
 
             if (securityProtocol.contains("SSL")) {
+                if (clientKeytoolCertificateGenerator != null && Path.of(clientKeytoolCertificateGenerator.getKeyStoreLocation()).toFile().exists()) {
+                    // SSL client auth case
+
+                    kafkaConfig.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, clientKeytoolCertificateGenerator.getKeyStoreLocation());
+                    kafkaConfig.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, clientKeytoolCertificateGenerator.getPassword());
+                    kafkaConfig.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, clientKeytoolCertificateGenerator.getPassword());
+                }
+
+                // this step is really only required if using self-signed.
                 try {
-                    clientKeytoolCertificateGenerator.generateTrustStore(brokerKeytoolCertificateGenerator.getCertFilePath(), "localhost");
+                    brokerKeytoolCertificateGenerator.generateTrustStore(brokerKeytoolCertificateGenerator.getCertFilePath(), "localhost");
```

Comment:
My intention here is to have client side certificates in a temporary folder and broker side certificates in different temporary folder to not override the truststore file. That's why I needed to initialize the clientKeytoolCertificateGenerator.

I believe that with your change, the use case where both need to add the certificates to its own truststore is not going to work because the alias in the truststore is overrided. So we would need to modify the alias to a different value than `localhost`

Reply:
+1 to renaming the alias.  on the client side "localhost" doesn't really make sense

"the use case where both need to add the certificates to the common truststore " - why would we need both certs in one truststore. There always a maximum of one, isn't there?

 ordinary ssl case:
* client needs the broker's cert in his trust store

ssl client auth
* client needs the broker's cert in his trust store
* broker needs client's cert in his trust store.

Reply:
Setting here the `brokerKeytoolCertificateGenerator` is going to insert the truststore with the broker cert (no key) in the broker folder, not in the client folder. That's why I needed to initialize the `clientKeytoolCertificateGenerator`.

I created two different variables to separate both truststore files, but using here the `brokerKeytoolCertificateGenerator` it is going to reuse the broker's one

Reply:
In the ordinary ssl case, you generate a key/cert for the broker only.   As we are using self signed certs the client needs the cert of the broker in its truststore.    The client doesn't require a key/cert of its own.  It does not make sense for the client to have a `clientKeytoolCertificateGenerator` in this case.

Reply:
That's the behavior of my implementation, the `clientKeytoolCertificateGenerator` is the tool that generates the key/cert for the client (not needed in that case), but also the truststore for the client. The `brokerKeytoolCertificateGenerator` makes the same for the broker. That's why the initialization of the instance is needed.

When calling the `generateTrustStore` method, the parameter is always requesting the cert file from the other actor: 

**clientKeytoolCertificateGenerator**.generateTrustStore(**brokerKeytoolCertificateGenerator**.getCertFilePath(), "localhost");
**brokerKeytoolCertificateGenerator**.generateTrustStore(**clientKeytoolCertificateGenerator**.getCertFilePath(), clientEndpoint.getConnect().getHost());

I think this is the correct behavior, but maybe having to create an instance of `clientKeytoolCertificateGenerator` in the test and pass it by parameter is not the best way to do it, but I didn't find a better way to do it.